### PR TITLE
ROX-23611: Force ebpf collection method for stackrox ebpf tests

### PR DESCRIPTION
**Pull request by JoukoVirtanen**
## Description

Currently CI tests with the EBPF collection method are being run with CORE_BPF instead. With this PR the EBPF collection method will be used in CI tests for both operator and helm chart based tests. 4.4 made it so that even if the collection method is set to EBPF, the collection method will be CORE_BPF, unless forceCollectionMethod is set to true. This was not done before.

This PR makes it so that

For helm charts
If the FORCE_COLLECTION_METHOD environment variable is set to true it will also be set to true in the helm charts when using the deployment scripts. This environment variable is set to true for CI.

For operator based CI tests
forceCollectionMethod is hard coded to true in the secured cluster CR yaml file.

The changes here should be backported to 4.4.

## Checklist
- [x] Investigated and inspected CI test results
~~- [ ] Unit test and regression tests added~~
~~- [ ] Evaluated and added CHANGELOG entry if required~~
~~- [ ] Determined and documented upgrade steps~~
~~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

If any of these don't apply, please comment below.

This PR only impacts internal testing, so does not require new tests or documentation.

### Here I tell how I validated my change

The changes in this PR were tested in an openshift/release PR here https://github.com/openshift/release/pull/52640 and here https://github.com/openshift/release/pull/52856

That PR is only for testing and the changes there do not have to reviewed.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: deploy/common/k8sbased.sh [hunk 1]
```diff
@@ -759,6 +759,11 @@ function launch_sensor {
         helm_chart="${SENSOR_CHART_DIR_OVERRIDE}"
       fi
 
+      if [[ "${FORCE_COLLECTION_METHOD:-false}" == "true" ]]; then
+	echo "Forcing collection method"
+	extra_helm_config+=(--set "collector.forceCollectionMethod=true")
```
Comment (JoukoVirtanen) [inline deploy/common/k8sbased.sh:764]: Fix indentation

**Review by ovalenti — APPROVED** [review]
Just a minor indentation to fix, otherwise this seems good to me !

**Review by gavin-stackrox — APPROVED** [review]
LGTM assuming ebpf tests are run and they do the right thing

**Review by gavin-stackrox — APPROVED** [review]

**Pull request merged**